installation issue of ansible i face in ubuntu 14.04 64 bit

##### ISSUE TYPE
Documentation Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION
2.2

##### SUMMARY
hi
i am facing installation issue for ansible i face in ubuntu 14.04 64 bit

```
[vagrant@db ~]$ sudo pip install ansible
DEPRECATION: Python 2.6 is no longer supported by the Python core team, please upgrade your Python. A future version of pip will drop support for Python 2.6
Collecting ansible
/usr/lib/python2.6/site-packages/pip-8.1.2-py2.6.egg/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/lib/python2.6/site-packages/pip-8.1.2-py2.6.egg/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Using cached ansible-2.1.0.0.tar.gz
Collecting paramiko (from ansible)
  Using cached paramiko-2.0.1-py2.py3-none-any.whl
Collecting jinja2 (from ansible)
  Using cached Jinja2-2.8-py2.py3-none-any.whl
Collecting PyYAML (from ansible)
  Using cached PyYAML-3.11.zip
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python2.6/site-packages (from ansible)
Collecting pycrypto>=2.6 (from ansible)
  Using cached pycrypto-2.6.1.tar.gz
Collecting cryptography>=1.1 (from paramiko->ansible)
  Using cached cryptography-1.4.tar.gz
Requirement already satisfied (use --upgrade to upgrade): pyasn1>=0.1.7 in /usr/lib/python2.6/site-packages (from paramiko->ansible)
Collecting MarkupSafe (from jinja2->ansible)
  Using cached MarkupSafe-0.23.tar.gz
Requirement already satisfied (use --upgrade to upgrade): idna>=2.0 in /usr/lib/python2.6/site-packages (from cryptography>=1.1->paramiko->ansible)
Requirement already satisfied (use --upgrade to upgrade): six>=1.4.1 in /usr/lib/python2.6/site-packages (from cryptography>=1.1->paramiko->ansible)
Requirement already satisfied (use --upgrade to upgrade): enum34 in /usr/lib/python2.6/site-packages (from cryptography>=1.1->paramiko->ansible)
Requirement already satisfied (use --upgrade to upgrade): ipaddress in /usr/lib/python2.6/site-packages (from cryptography>=1.1->paramiko->ansible)
Requirement already satisfied (use --upgrade to upgrade): cffi>=1.4.1 in /usr/lib64/python2.6/site-packages (from cryptography>=1.1->paramiko->ansible)
Requirement already satisfied (use --upgrade to upgrade): pycparser in /usr/lib/python2.6/site-packages (from cffi>=1.4.1->cryptography>=1.1->paramiko->ansible)
Installing collected packages: cryptography, paramiko, MarkupSafe, jinja2, PyYAML, pycrypto, ansible
  Running setup.py install for cryptography ... error
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ZgAqU0/cryptography/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-md8Td_-record/install-record.txt --single-version-externally-managed --compile:
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    compiling '_configtest.c':
    __thread int some_threadlocal_variable_42;
    c/_cffi_backend.c:15:17: error: ffi.h: No such file or directory
    In file included from c/_cffi_backend.c:63:
    c/malloc_closure.h:81: error: expected specifier-qualifier-list before ‘ffi_closure’
    c/malloc_closure.h: In function ‘more_core’:
    c/malloc_closure.h:117: warning: division by zero
    c/malloc_closure.h:149: error: ‘union mmaped_block’ has no member named ‘next’
    c/malloc_closure.h: At top level:
    c/malloc_closure.h:158: error: expected ‘)’ before ‘*’ token
    c/malloc_closure.h:166: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘*’ token
    c/_cffi_backend.c:256: error: expected specifier-qualifier-list before ‘ffi_cif’
    c/_cffi_backend.c: In function ‘cdataowninggc_dealloc’:
    c/_cffi_backend.c:1612: error: ‘ffi_closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1612: error: (Each undeclared identifier is reported only once
    c/_cffi_backend.c:1612: error: for each function it appears in.)
    c/_cffi_backend.c:1612: error: ‘closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1612: error: expected expression before ‘)’ token
    c/_cffi_backend.c:1615: warning: implicit declaration of function ‘cffi_closure_free’
    c/_cffi_backend.c: In function ‘cdataowninggc_traverse’:
    c/_cffi_backend.c:1632: error: ‘ffi_closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1632: error: ‘closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1632: error: expected expression before ‘)’ token
    c/_cffi_backend.c: In function ‘cdataowninggc_clear’:
    c/_cffi_backend.c:1653: error: ‘ffi_closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1653: error: ‘closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1653: error: expected expression before ‘)’ token
    c/_cffi_backend.c: In function ‘cdataowninggc_repr’:
    c/_cffi_backend.c:1841: error: ‘ffi_closure’ undeclared (first use in this function)
    c/_cffi_backend.c:1841: error: expected expression before ‘)’ token
    c/_cffi_backend.c: At top level:
    c/_cffi_backend.c:2442: error: expected declaration specifiers or ‘...’ before ‘ffi_abi’
    c/_cffi_backend.c: In function ‘cdata_call’:
    c/_cffi_backend.c:2571: error: ‘ffi_abi’ undeclared (first use in this function)
    c/_cffi_backend.c:2571: error: expected ‘;’ before ‘fabi’
    c/_cffi_backend.c:2613: error: ‘fabi’ undeclared (first use in this function)
    c/_cffi_backend.c:2617: error: too many arguments to function ‘fb_prepare_cif’
    c/_cffi_backend.c:2622: error: ‘cif_description_t’ has no member named ‘exchange_size’
    c/_cffi_backend.c:2632: error: ‘cif_description_t’ has no member named ‘exchange_offset_arg’
    c/_cffi_backend.c:2662: error: ‘cif_description_t’ has no member named ‘exchange_offset_arg’
    c/_cffi_backend.c:2667: warning: implicit declaration of function ‘ffi_call’
    c/_cffi_backend.c:2667: error: ‘cif_description_t’ has no member named ‘cif’
    c/_cffi_backend.c: In function ‘new_primitive_type’:
    c/_cffi_backend.c:3813: error: ‘ffi_type’ undeclared (first use in this function)
    c/_cffi_backend.c:3813: error: ‘ffitype’ undeclared (first use in this function)
    c/_cffi_backend.c:3831: error: ‘ffi_type_sint8’ undeclared (first use in this function)
    c/_cffi_backend.c:3832: error: ‘ffi_type_sint16’ undeclared (first use in this function)
    c/_cffi_backend.c:3833: error: ‘ffi_type_sint32’ undeclared (first use in this function)
    c/_cffi_backend.c:3834: error: ‘ffi_type_sint64’ undeclared (first use in this function)
    c/_cffi_backend.c:3840: error: ‘ffi_type_float’ undeclared (first use in this function)
    c/_cffi_backend.c:3842: error: ‘ffi_type_double’ undeclared (first use in this function)
    c/_cffi_backend.c:3850: error: ‘ffi_type_longdouble’ undeclared (first use in this function)
    c/_cffi_backend.c:3857: error: ‘ffi_type_uint8’ undeclared (first use in this function)
    c/_cffi_backend.c:3858: error: ‘ffi_type_uint16’ undeclared (first use in this function)
    c/_cffi_backend.c:3859: error: ‘ffi_type_uint32’ undeclared (first use in this function)
    c/_cffi_backend.c:3860: error: ‘ffi_type_uint64’ undeclared (first use in this function)
    c/_cffi_backend.c: At top level:
    c/_cffi_backend.c:4496: error: expected specifier-qualifier-list before ‘ffi_type’
    c/_cffi_backend.c:4515: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘*’ token
    c/_cffi_backend.c: In function ‘fb_build’:
    c/_cffi_backend.c:4655: error: ‘struct funcbuilder_s’ has no member named ‘atypes’
    c/_cffi_backend.c:4655: error: ‘ffi_type’ undeclared (first use in this function)
    c/_cffi_backend.c:4655: error: expected expression before ‘)’ token
    c/_cffi_backend.c:4656: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4659: error: ‘struct funcbuilder_s’ has no member named ‘rtype’
    c/_cffi_backend.c:4659: warning: implicit declaration of function ‘fb_fill_type’
    c/_cffi_backend.c:4667: error: ‘cif_description_t’ has no member named ‘exchange_offset_arg’
    c/_cffi_backend.c:4670: error: ‘struct funcbuilder_s’ has no member named ‘rtype’
    c/_cffi_backend.c:4671: error: ‘ffi_arg’ undeclared (first use in this function)
    c/_cffi_backend.c:4681: error: ‘atype’ undeclared (first use in this function)
    c/_cffi_backend.c:4694: error: ‘struct funcbuilder_s’ has no member named ‘atypes’
    c/_cffi_backend.c:4695: error: ‘struct funcbuilder_s’ has no member named ‘atypes’
    c/_cffi_backend.c:4698: error: ‘cif_description_t’ has no member named ‘exchange_offset_arg’
    c/_cffi_backend.c:4707: error: ‘cif_description_t’ has no member named ‘exchange_size’
    c/_cffi_backend.c: In function ‘fb_build_name’:
    c/_cffi_backend.c:4731: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4743: error: ‘struct funcbuilder_s’ has no member named ‘fct’
    c/_cffi_backend.c:4746: error: ‘struct funcbuilder_s’ has no member named ‘fct’
    c/_cffi_backend.c: In function ‘fb_prepare_ctype’:
    c/_cffi_backend.c:4790: error: ‘struct funcbuilder_s’ has no member named ‘fct’
    c/_cffi_backend.c:4807: error: ‘struct funcbuilder_s’ has no member named ‘fct’
    c/_cffi_backend.c: At top level:
    c/_cffi_backend.c:4827: error: expected declaration specifiers or ‘...’ before ‘ffi_abi’
    c/_cffi_backend.c: In function ‘fb_prepare_cif’:
    c/_cffi_backend.c:4854: warning: implicit declaration of function ‘ffi_prep_cif’
    c/_cffi_backend.c:4854: error: ‘cif_description_t’ has no member named ‘cif’
    c/_cffi_backend.c:4854: error: ‘fabi’ undeclared (first use in this function)
    c/_cffi_backend.c:4854: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4855: error: ‘struct funcbuilder_s’ has no member named ‘rtype’
    c/_cffi_backend.c:4855: error: ‘struct funcbuilder_s’ has no member named ‘atypes’
    c/_cffi_backend.c:4855: error: ‘FFI_OK’ undeclared (first use in this function)
    c/_cffi_backend.c: In function ‘new_function_type’:
    c/_cffi_backend.c:4899: error: too many arguments to function ‘fb_prepare_cif’
    c/_cffi_backend.c:4913: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4923: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4933: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4936: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4937: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c:4939: error: ‘struct funcbuilder_s’ has no member named ‘nargs’
    c/_cffi_backend.c: In function ‘b_new_function_type’:
    c/_cffi_backend.c:4950: error: ‘FFI_DEFAULT_ABI’ undeclared (first use in this function)
    c/_cffi_backend.c: In function ‘convert_from_object_fficallback’:
    c/_cffi_backend.c:4970: error: ‘ffi_arg’ undeclared (first use in this function)
    c/_cffi_backend.c: At top level:
    c/_cffi_backend.c:5152: error: expected ‘)’ before ‘*’ token
    c/_cffi_backend.c: In function ‘prepare_callback_info_tuple’:
    c/_cffi_backend.c:5194: error: ‘ffi_arg’ undeclared (first use in this function)
    c/_cffi_backend.c: In function ‘b_callback’:
    c/_cffi_backend.c:5227: error: ‘ffi_closure’ undeclared (first use in this function)
    c/_cffi_backend.c:5227: error: ‘closure’ undeclared (first use in this function)
    c/_cffi_backend.c:5237: warning: implicit declaration of function ‘cffi_closure_alloc’
    c/_cffi_backend.c:5255: warning: implicit declaration of function ‘ffi_prep_closure’
    c/_cffi_backend.c:5255: error: ‘cif_description_t’ has no member named ‘cif’
    c/_cffi_backend.c:5256: error: ‘invoke_callback’ undeclared (first use in this function)
    c/_cffi_backend.c:5256: error: ‘FFI_OK’ undeclared (first use in this function)
    In file included from c/cffi1_module.c:3,
                     from c/_cffi_backend.c:6636:
    c/realize_c_type.c: In function ‘realize_c_type_or_func’:
    c/realize_c_type.c:577: error: ‘FFI_DEFAULT_ABI’ undeclared (first use in this function)
    c/_cffi_backend.c: In function ‘init_cffi_backend’:
    c/_cffi_backend.c:6728: error: ‘FFI_DEFAULT_ABI’ undeclared (first use in this function)
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ZgAqU0/cryptography/setup.py", line 333, in <module>
        **keywords_with_side_effects(sys.argv)
      File "/usr/lib64/python2.6/distutils/core.py", line 113, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 221, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 245, in fetch_build_eggs
        parse_requirements(requires), installer=self.fetch_build_egg
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 538, in resolve
        dist = best[req.key] = env.best_match(req, self, installer)
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 780, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 792, in obtain
        return installer(requirement)
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 293, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 472, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 502, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 681, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 958, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 949, in run_setup
        raise DistutilsError("Setup script exited with %s" % (v.args[0],))
    distutils.errors.DistutilsError: Setup script exited with error: command 'gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ZgAqU0/cryptography/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-md8Td_-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-ZgAqU0/cryptography/

```

--- a/docs/docsite/rst/installation_guide/intro_installation.rst
+++ b/docs/docsite/rst/installation_guide/intro_installation.rst
@@ -159,7 +159,7 @@ Debian/Ubuntu packages can also be built from the source checkout, run:
 
     $ make deb
 
-You may also wish to run from source to get the latest, which is covered above.
+You may also wish to run from source to get the latest, which is covered below.
 
 Latest Releases Via Apt (Debian)
 ++++++++++++++++++++++++++++++++
@@ -395,6 +395,8 @@ bugs and feature ideas.
        Examples of basic commands
    :ref:`working_with_playbooks`
        Learning ansible's configuration management language
+   :ref:`installation_faqs`
+       Ansible Installation related to FAQs
    `Mailing List <http://groups.google.com/group/ansible-project>`_
        Questions? Help? Ideas?  Stop by the list on Google Groups
    `irc.freenode.net <http://irc.freenode.net>`_

--- a/docs/docsite/rst/reference_appendices/faq.rst
+++ b/docs/docsite/rst/reference_appendices/faq.rst
@@ -131,6 +131,19 @@ requires Python 2, you can also report a bug on our `bug tracker
 
 Do not replace the shebang lines of your python modules.  Ansible will do this for you automatically at deploy time.
 
+.. _installation_faqs:
+
+How do I handle the package dependencies required by Ansible package dependencies during Ansible installation ?
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+While installing Ansible, sometimes you may encounter errors such as `No package 'libffi' found` or `fatal error: Python.h: No such file or directory`
+These errors are generally caused by the missing packages which are dependencies of the packages required by Ansible.
+For example, `libffi` package is dependency of `pynacl` and `paramiko` (Ansible -> paramiko -> pynacl -> libffi).
+
+In order to solve these kinds of dependency issue, you may need to install required packages using the OS native package managers (e.g., `yum`, `dnf` or `apt`) or as mentioned in the package installation guide.
+
+Please refer the documentation of the respective package for such dependencies and their installation methods.
+
 Common Platform Issues
 ++++++++++++++++++++++
 

